### ISSUE TITLE
pacaur doesn't find makedepends if PKGDEST variable is set

### ISSUE THREAD

**Issue by abouvier** [rmarquis/pacaur#535]
pacaur doesn't find (and so doesn't install) AUR **makedepends** packages when installing a package, if the variable PKGDEST is set to a non empty value. There is no problem with AUR **depends** packages.

Example with php-cs-fixer who requires php-box for compiling:

Working:
```
$ PKGDEST= pacaur -S php-cs-fixer
:: Package(s) php-cs-fixer not found in repositories, trying AUR...
:: resolving dependencies...
:: looking for inter-conflicts...

AUR Packages  (2)  Old Version  New Version

aur/php-box                     2.7.4-1
aur/php-cs-fixer                1.11.6-1

:: Proceed with installation? [Y/n]
```
Not working:
```
$ PKGDEST=/tmp pacaur -S php-cs-fixer
:: Package(s) php-cs-fixer not found in repositories, trying AUR...
:: resolving dependencies...
:: looking for inter-conflicts...

AUR Packages  (1)  Old Version  New Version

aur/php-cs-fixer                1.11.6-1

:: Proceed with installation? [Y/n]
```

**Comment by rmarquis** [comment]
Thanks for the report, I'll take a look asap. This looks indeed weird.

**Comment by rmarquis** [comment]
Looks like the issue has been introduced in 1f520e625647f30600acd680e3ea89f98e98a704 more than a year ago (v4.2.25).

It seems to happen only for the top level package, makedeps of dependencies are actually correctly taken into account. This might explain why that regression hasn't been discovered till yet.

**Comment by rmarquis** [comment]
Fixed in 96b67989c0c5a19520be6650c921fc154d829e5c. The cache now independenctly checks every makedeps and checkdeps, so the fix implemented in #326 is still fully functional.

The `--foreign` flag must still be adjusted for this new behavior.

**Comment by abouvier** [comment]
Plot twist, needed makedepends who are not installed but in the cache, are not installed when installing/updating a package.

Example of installing `php-cs-fixer` with the needed makedepend `php-box` currently not installed, but presents in the cache:

Not working (regardless of the value of `PKGDEST`):
```
$ pacaur -S php-cs-fixer
:: Package(s) php-cs-fixer not found in repositories, trying AUR...
:: resolving dependencies...
:: looking for inter-conflicts...

AUR Packages  (1)  Old Version  New Version

aur/php-cs-fixer                1.12.0-1

:: Proceed with installation? [Y/n]
```
Working with `--foreign` (not sure why):
```
$ pacaur -S php-cs-fixer --foreign
:: Package(s) php-cs-fixer not found in repositories, trying AUR...
:: resolving dependencies...
:: looking for inter-conflicts...

AUR Packages  (2)  Old Version  New Version

aur/php-box                     2.7.4-1           (cached)
aur/php-cs-fixer                1.12.0-1

:: Proceed with installation? [Y/n]
```
Working if the cached package is removed:
```
$ rm ~/.cache/pacman/pkg/php-box-2.7.4-1-any.pkg.tar.xz
$ pacaur -S php-cs-fixer
:: Package(s) php-cs-fixer not found in repositories, trying AUR...
:: resolving dependencies...
:: looking for inter-conflicts...

AUR Packages  (2)  Old Version  New Version

aur/php-box                     2.7.4-1
aur/php-cs-fixer                1.12.0-1

:: Proceed with installation? [Y/n]
```
Or maybe I didn't understood the `--foreign` option and everything above is normal :p

**Comment by rmarquis** [comment]
Yep, you're right. I thought I had properly tested this user case, but it's still buggy.

The ```--foreign``` flag rebuilds all the AUR dependencies, even if some of these are already satisfied. Its usefullness isn't really demonstrated in this simple case, but it is working as intended.

**Comment by rmarquis** [comment]
Quick fixed in 2b79323bd3ddbaa6eb3712fe5bfb1fc133187fc5.

The individual cache check for makedeps/checkdeps is now broken (it's all or nothing), but I'll fix that later on as this is a more urgent fix. Closing here, reopening #326.
